runners/spark/src/main/java/org/apache/beam/runners/spark/metrics/SparkMetricsContainer.java

```diff
@@ -120,6 +128,15 @@ private void updateDistributions(Iterable<MetricUpdate<DistributionData>> update
     }
   }
 
+  private void updateGauges(Iterable<MetricUpdate<GaugeData>> updates) {
+    for (MetricUpdate<GaugeData> update : updates) {
+      MetricKey key = update.getKey();
+      MetricUpdate<GaugeData> current = gauges.get(key);
+      gauges.put(key, current != null
+          ? MetricUpdate.create(key, current.getUpdate().combine(update.getUpdate())) : update);
```

Comment:
I completely agree.
We have indeed considered this and are planning to extract such an interface. 
Work on this is tracked in [BEAM-1672](https://issues.apache.org/jira/browse/BEAM-1672)